crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -2139,6 +2145,37 @@ impl actor::HcP2p for HolochainP2pActor {
                 .collect())
         })
     }
+
+    fn conductor_db_getter(&self) -> GetDbConductor {
+        self.blocks_db_getter.clone()
+    }
+
+    fn block(
+        &self,
+        block_target: holochain_zome_types::block::BlockTarget,
+    ) -> BoxFut<'_, HolochainP2pResult<()>> {
+        use holochain_timestamp::Timestamp;
+        use holochain_zome_types::block::BlockTarget;
+
+        Box::pin(async move {
+            let db = self.conductor_db_getter()().await;
+            match block_target {
+                BlockTarget::Cell(cell_id, cell_block_reason) => block(
+                    &db,
+                    Block::new(
+                        BlockTarget::Cell(cell_id, cell_block_reason),
+                        InclusiveTimestampInterval::try_new(Timestamp::now(), Timestamp::MAX)
+                            .map_err(|err| HolochainP2pError::other(format!("Could not create a timestamp interval while writing a block: {err}")))?,
+                    ),
+                )
+                .await
+                .map_err(|err| HolochainP2pError::other(format!("Could not write block to database: {err}"))),
```

Comment:
After blocking an agent, they should also be removed from the peer store. I'll do that in a follow-up PR.